components/brave_ads/core/internal/creatives/notification_ads/creative_notification_ad_info.cc

```diff
@@ -16,7 +16,7 @@ CreativeNotificationAdInfo::CreativeNotificationAdInfo(
 bool CreativeNotificationAdInfo::operator==(
     const CreativeNotificationAdInfo& other) const {
   return CreativeAdInfo::operator==(other) && title == other.title &&
-         body == other.body;
+         body == other.body && embedding == other.embedding;
```

Comment:
embedding should not be checked in this subclassed object. Please move the check to the base AdInfo object.